### PR TITLE
[#2950] - Limit columns when working with RQG

### PR DESCRIPTION
**Pull request by marvinkome**
When doing Combine columns or generate geopoint transformations,
This commit add the columns logic when working with RQG or not.
The logic can be found in the connected issue

- [ ] **Update release notes if necessary**

### CODE REVIEW

File: client/src/components/dataset/sidebars/GenerateGeopoints.jsx [hunk 1]
```diff
@@ -54,6 +55,27 @@ export default class GenerateGeopoints extends Component {
     this.handleChangeColumnTitleGeo = this.handleChangeColumnTitleGeo.bind(this);
   }
 
+  getColumns(latOrLong) {
```
Comment (iperdomo) [inline client/src/components/dataset/sidebars/GenerateGeopoints.jsx:58]: I guess this logic can be unit tested? it is possible?

Comment (marvinkome) [reply client/src/components/dataset/sidebars/GenerateGeopoints.jsx:58]: I'd have to extract it out to a separate function to unit test it. Or else I'd be testing the whole component together

File: client/src/utilities/column.js [hunk 2]
```diff
@@ -97,3 +97,22 @@ export const columnSelectOptions = (intl, columns) => {
   const res = extractColumnOptions(columns);
   return isImmutable(res) ? res.toArray() : res;
 };
+
+// For combine columns and generate geopoints column select
```
Comment (tangrammer) [inline client/src/utilities/column.js:101]: will we need this comment on mid-long term?

File: client/src/utilities/column.js [hunk 3]
```diff
@@ -97,3 +97,22 @@ export const columnSelectOptions = (intl, columns) => {
...
+export const columnBasedOnOppositeColumn = (columns, oppositeColumnName) => {
```
Comment (tangrammer) [inline client/src/utilities/column.js:102]: perhaps we could find a better name for this function 🤔 . The logic inside doesn't seem to be represented by the name
My proposal and refactor will be:
```js
export const filterColumnsByDataGroupDimension = (columns, columnName) => {
  // if column is not selected the return all columns
  if (!columnName) {
    return columns;
  }

  const column = columns.find(col => col.get('columnName') === columnName);

  // if column is not repeatable return all not repeatable columns
  if (!column.get('repeatable')) {
    return columns.filter(col => !col.get('repeatable'));
  }

  // if column is repeatable, return only columns from the same group
  return columns.filter(col => col.get('groupName') === column.get('groupName'));
};

```

**Review by tangrammer — CHANGES_REQUESTED** [review]
do you aggree with my proposal
refactor?

**Review by tangrammer — APPROVED** [review]

**Pull request merged**